sample/SipInterconnect/README.md

```diff
@@ -0,0 +1,238 @@
+
+* Use TLS and enable secure calls (SRTP) for signaling to avoid the possibility of
+  intercepting the communications.
+
+* Enable SIP authentication on your server. Otherwise, anyone who knows your SIP URI could
+  send calls to your server.
+
+If required, you can also block the traffic not coming from the IP addresses of the OpenTok
+SIP gateway, which uses the following IP blocks:
+
+* 52.200.60.16 &ndash; 52.200.60.31
+* 52.41.63.240 &ndash; 52.41.63.255
+* 52.51.63.16 &ndash; 52.51.63.31
+
+Be aware that TokBox can add new IP blocks in the future, and you
+will need to change your configuration.
+
+### Technical details
+
+**RFC3550 (RTP/RTCP) support:** Media traffic can be encrypted (SRTP) or non-­encrypted
+(plain RTP). In case of encryption, both DTLS and SDES protocols are supported.
+
+**Codec support:** The OpenTok SIP gateway supports the OPUS, G.711, and G.722 audio codecs.
+
+**Signaling:** The OpenTok SIP gateway supports RFC3561 (SIP) over UDP, TCP, and TLS.
+Contact TokBox if you need information or support for any specific extension.
+
+The OpenTok SIP gateway will not accept any SIP message coming from the a third-party SIP
+platform unless it is part of a SIP dialog initiated by the OpenTok SIP gateway.
+Calls initiated with the OpenTok SIP gateway can be put on ­hold using either a `re-­INVITE`
+with the `sendonly/inactive` direction in the SDP or a `re-­INVITE` with port 0 in the SDP.
+
+**Other considerations:** Early media is disabled, and DTMFs are not currently supported.
+
+
+### Additional Information
+
+For additional information see [SIP Interconnect](https://www.tokbox.com/developer/guides/sip/) at the TokBox Developer Center. 
+
+
+
+## Prerequisites
+
+To be prepared to develop your OpenTok SIP Interconnect app:
+
+1. Review the [OpenTok.js](https://tokbox.com/developer/sdks/js/) requirements.
+2. Your app will need **Token** and **API Key**, which you can get at the [OpenTok Developer Dashboard](https://dashboard.tokbox.com/). Set the API Key and API Secret in [config.js](./config.js).
+
+
+To install the OpenTok SIP Interconnect Sample App, run the following commands:
+
+```
+npm install
+node bin/www
+```
+
+
+_**IMPORTANT:** In order to deploy an OpenTok SPI Interconnect app, your web domain must use HTTPS._
+
+
+## Quick start
+
+The web page that loads the sample app for JavaScript must be served over HTTP/HTTPS. Browser security limitations prevent you from publishing video using a `file://` path, as discussed in the OpenTok.js [Release Notes](https://www.tokbox.com/developer/sdks/js/release-notes.html#knownIssues). To support clients running [Chrome 47 or later](https://groups.google.com/forum/#!topic/discuss-webrtc/sq5CVmY69sc), HTTPS is required. A web server such as [MAMP](https://www.mamp.info/) or [XAMPP](https://www.apachefriends.org/index.html) will work, or you can use a cloud service such as [Heroku](https://www.heroku.com/) to host the application.
+
+
+
+## Exploring the code
```

Comment:
you should leverage the Hello World sample and assume the reader has figured that out. See the Archiving sample for how we did this there.